base/broadcast.jl

```diff
@@ -534,10 +534,7 @@ end
 # explicit calls to view.   (All of this can go away if slices
 # are changed to generate views by default.)
 
-Base.@propagate_inbounds dotview(args...) = getindex(args...)
-Base.@propagate_inbounds dotview(A::AbstractArray, args...) = view(A, args...)
-Base.@propagate_inbounds dotview(A::AbstractArray{<:AbstractArray}, args::Integer...) = getindex(A, args...)
-
+Base.@propagate_inbounds dotview(args...) = Base.maybeview(args...)
```

Comment:
With this `dotview` wouldn't be needed and could also be deprecated in favor of `maybeview`, right?

(Though it would require changing [this line](https://github.com/JuliaLang/julia/blob/0c1edbf3904a84a7e438fe729148279717b353cd/src/julia-syntax.scm#L1717) in the parser)

Reply:
For now the two must be uncoupled in order to have the deprecation... and it's probably safest to keep them separate in general.